### PR TITLE
Add configs, owner, hypervisor_id, http_proxy virtwho_conf cases

### PR DESCRIPTION
**Pull request by hkx303**
**Description**
This PR deponds on #70 , mark it draft PR

- [X] test_configs_in_virtwho_conf
- [X] test_owner_in_virtwho_conf
- [X] test_hypervisor_id_in_virtwho_conf
- [X] test_http_proxy_in_virtwho_conf

Test Result
All passed locally

### CODE REVIEW

File: tests/function/test_config.py [hunk 1]
```diff
@@ -190,3 +192,309 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
...
+            2. Run virt-who with hypervisor_id=uuid in /etc/virt-who.conf
+            3. Run virt-who with hypervisor_id=hwuuid in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. Succeeded to run the virt-who, the hypervisor_id in mapping info should be hostname
+            2. Succeeded to run the virt-who, the hypervisor_id in mapping info should be uuid
+            3. Succeeded to run the virt-who, the hypervisor_id in mapping info should be hwuuid
+        """
+        host_uuid = hypervisor_handler.esx_uuid
```
Comment (Junefen) [inline tests/function/test_config.py:389]: This test file will be run for the main hypervisor, but not only the `esx` mode, we can select which one or ones will be the main hypervisors.
So this is why I replace the `hypervisor_handler` by `hypervisor_data` fixture in conftest

File: tests/function/test_config.py [hunk 2]
```diff
@@ -190,3 +192,309 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
...
+
+        :expectedresults:
+
+            1. Succeeded to run the virt-who and ignore the configurations files in
+            /etc/virt-who.d/ dir
+        """
+        config_file = '/tmp/test_config_configs.conf'
+        guest_uuid = hypervisor_handler.guest_uuid
+        globalconf.update('global', 'debug', 'True')
+        ssh_host.runcmd(f'\\cp -f {HYPERVISOR_FILE} {config_file}')
```
Comment (Junefen) [inline tests/function/test_config.py:329]: Could we define a public base function for request.

Comment (Junefen) [reply tests/function/test_config.py:331]: Not a necessary request, close it now.

File: tests/function/test_config.py [hunk 3]
```diff
@@ -301,3 +306,197 @@ def test_log_dir_and_log_file_in_virtwho_conf(self, virtwho, globalconf, hypervi
...
+
+        globalconf.update('system_environment', 'no_proxy', '*')
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1)
+
+        globalconf.delete('system_environment')
+
+        globalconf.update('system_environment', 'https_proxy', https_proxy_url)
```
Comment (Junefen) [inline tests/function/test_config.py:481]: This may also need a loop.

File: tests/function/test_config.py (hunk 1 above)
Comment (hkx303) [reply tests/function/test_config.py]: updated

File: tests/function/test_config.py [hunk 4]
```diff
@@ -190,3 +194,309 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
...
+
+        :expectedresults:
+
+            1. Succeeded to run the virt-who with owner setting
+        """
+        guest_uuid = hypervisor_data['guest_uuid']
+        owner = register_data['default_org']
+        globalconf.update('global', 'debug', 'True')
+
+        globalconf.update('defaults', 'owner', owner)
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/function/test_config.py]: - removed
- update the wrong value owner

File: tests/function/test_config.py [hunk 5]
```diff
@@ -190,3 +194,309 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
...
+        if HYPERVISOR in ['esx', 'rhevm']:
+            host_hwuuid = hypervisor_data['hypervisor_hwuuid']
+            globalconf.update('defaults', 'hypervisor_id', 'hwuuid')
+            result = virtwho.run_service()
+            assert (result['error'] == 0
+                    and result['send'] == 1
+                    and result['thread'] == 1
+                    and result['hypervisor_id'] == host_hwuuid)
+
+        #if register is stage, do we need to clean the consumer?
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/function/test_config.py]: added

File: tests/function/test_config.py [hunk 6]
```diff
@@ -190,3 +194,309 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
...
+            2. Succeeded to run the virt-who, the hypervisor_id in mapping info should be uuid
+            3. Succeeded to run the virt-who, the hypervisor_id in mapping info should be hwuuid
+        """
+        host_uuid = hypervisor_data['hypervisor_uuid']
+        host_name = hypervisor_data['hypervisor_hostname']
+        globalconf.update('global', 'debug', 'True')
+        # we default have the hypervisor_id section in the config file in /etc/virt-who.d/
+        hypervisor.delete('hypervisor_id')
+
+        globalconf.update('defaults', 'hypervisor_id', 'hostname')
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/function/test_config.py]: sure, updated

File: tests/function/test_config.py [hunk 7]
```diff
@@ -190,3 +194,309 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
...
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1
+                and connection_msg in result['log']
+                and proxy_msg in result['log'])
+
+        globalconf.update('system_environment', 'http_proxy', unreach_http_proxy)
+        result = virtwho.run_service()
+        assert (result['error'] == 1 or 2)
+        error_msgs = ['Connection refused', 'Cannot connect to proxy', 'Connection timed out',
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/function/test_config.py]: updated, the same as owner

File: tests/function/test_config.py (hunk 3 above)
Comment (hkx303) [reply tests/function/test_config.py]: updated

File: tests/function/test_config.py [hunk 8]
```diff
@@ -301,3 +306,172 @@ def test_log_dir_and_log_file_in_virtwho_conf(self, virtwho, globalconf, hypervi
...
+        for hypervisor_id in hypervisor_ids:
+            # only esx and rhevm modes support hwuuid
+            globalconf.update('defaults', 'hypervisor_id', hypervisor_id)
+            result = virtwho.run_service()
+            assert (result['error'] == 0
+                    and result['send'] == 1
+                    and result['thread'] == 1
+                    and result['hypervisor_id'] == hypervisor_data[f'hypervisor_{hypervisor_id}'])
+
+        if REGISTER is 'rhsm':
```
Comment (Junefen) [inline tests/function/test_config.py:420]: - If this `if` branch should be in the `for` loop?
- And as discussion we should define new functions to make sure the hypervisor entry can be shown successfully in the register server web and for satellite there should only one entry existing.

File: tests/function/test_config.py [hunk 9]
```diff
@@ -301,3 +306,175 @@ def test_log_dir_and_log_file_in_virtwho_conf(self, virtwho, globalconf, hypervi
...
+            assert (result['error'] == 0
+                    and result['send'] == 1
+                    and result['thread'] == 1
+                    and result['hypervisor_id'] == hypervisor_data[f'hypervisor_{hypervisor_id}'])
+            if REGISTER == 'rhsm':
+                assert rhsm.consumers(hypervisor_data['hypervisor_hostname'])
+                rhsm.delete(hypervisor_data['hypervisor_hostname'])
+            else:
+                assert satellite.host_id(hypervisor_data[f'hypervisor_{hypervisor_id}'])
+                satellite.host_delete(hypervisor_data[f'hypervisor_{hypervisor_id}'])
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/function/test_config.py]: updated

**Pull request merged**